microsoft-edge/webview2/how-to/webresourcerequested.md

```diff
@@ -0,0 +1,582 @@
+
+This article describes how you can modify network requests.  Use this API and approach to:
+* Upload local file content to your app to add support for offline functionality.
+* Block content in a webpage, such as specific images.
+* Fine-tune authentication for specific pages.
+
+**Terminology:**
+
+| Term | Definition |
+|---|---|
+| intercept | Your host app can intercept (receive) a request that is sent from the WebView2 control to the HTTP server, read or modify the request, and then send the unchanged or modified request to the HTTP server (or to local code instead of the HTTP server). |
+| override | Your host app can override (ignore) a response that's sent from the HTTP server to the WebView2 control, and send a custom response to the WebView2 control instead of the original response. |
+
+
+<!-- ====================================================================== -->
+## When to use custom vs. basic approaches
+
+The `webResourceRequested` and `webResourceResponseReceived` events are powerful tools for various scenarios.  This is a low-level API that gives more control, but requires more coding and is complicated to use.  For some common scenarios, we provide APIs that are easier to use and are optimized for those specific scenarios, and we recommend you use those rather than the APIs discussed in this article.
+
+Instead of using the WebResourceRequested APIs, you can use these other approaches that are build on top of `WebResourceRequested` and related APIs:
+* [Basic Authentication](/microsoft-edge/webview2/concepts/basic-authentication?tabs=csharp)
+* [General navigation](/microsoft-edge/webview2/concepts/navigation-events) 
+* [Managing cookies in WebView2](/microsoft-edge/webview2/reference/win32/icorewebview2)
+
+**Note:** If you are using virtually mapped URLs, using the `WebResourceRequested` event isn't supported.  This is because the `WebResourceRequested` event isn't fired for 
+the [SetVirtualHostNameToFolderMapping method](/microsoft-edge/webview2/reference/win32/icorewebview2_3#setvirtualhostnametofoldermapping).<!-- ClearVirtualHostNameToFolderMapping -->
+
+
+### How your host app, the WebView2 control, and the HTTP server interact
+
+The WebView2 control sits in between your host app and the HTTP server.  When your host app navigates to a URI, the WebView2 control sends a request to the HTTP server.  The HTTP server then sends a response to the WebView2 control.
+
+<!-- lexicon for the 3 actors:
+the HTTP server
+the WebView2 control
+the host app
+-->
+
+
+<!-- ====================================================================== -->
+## Intercepting a request (to monitor or modify it)
+
+Your host app can _intercept_ (receive) a request that is sent from the WebView2 control to the HTTP server, read or modify the request, and then send the unchanged or modified request to the HTTP server (or to local code instead of the HTTP server). 

```

Comment:
```suggestion
Your host app can _intercept_ a request that is sent from the WebView2 control to the HTTP server, read or modify the request, and then send the unchanged or modified request to the HTTP server (or to local code instead of the HTTP server). 

```